### ISSUE TITLE
Including structs straight up doesn't work ?

### ISSUE THREAD

**Issue by TcePrepK** [walt-grace/glsl-plugin-idea#38]
Just having a struct within the file, lets say
```glsl
struct TestStruct {
    float a;
    float b;
};

TestStruct test = TestStruct(0.0, 0.0); // Works completely fine!
```
works perfectly as expected... But when you move the struct definition to, lets say "structFile.glsl" and try to include, it doesn't seem to recognize the struct at all!
```glsl
#include "structFile.glsl"

TestStruct test = TestStruct(0.0, 0.0); // No type for the test, doesn't get colored etc.
```
I remember these working completely fine a few updates ago (1.0.74), but that version used to crash a lot of times basically forcing me to close/reopen my IDE, due to me working with glsl almost all the time

**Comment by walt-grace** [comment]
Hi @TcePrepK 
thanks for the valuable info. I'll look at this and the other bug and these should not be hard to fix.

There was a previous implementation of #include which was, as you've already noticed, pretty bad and made the IDE crash (probably due to import cycles). Now there is a new implementation that should work much better. 

I'll keep you updated how it's going.

**Comment by TcePrepK** [comment]
yeah, it was due to import cycles. You could see it indexing at the bottom of the idea indefinitely and crashing lol. It is good to hear they are an easy fix, your plugin is an actual life savior for me right now while I am constantly working with glsl! Ty for the quick response and fix!!!
